src/number.cr

```diff
@@ -239,6 +239,29 @@ struct Number
         @current += @step
       end
     end
+
+    # Overrides `Enumerable#sum` to use more performant implementation on integer
+    # ranges.
+    def sum(initial)
```

Comment:
Is there an existing spec for this?

Reply:
Yes, I just copied this method from `Range::StepIterator` because the sum specs in `range_spec` happen to call this method. Especially these specs runs *very* long without it:

https://github.com/crystal-lang/crystal/blob/cb09b0e218768c759f261424f10a0abb48edff0f/spec/std/range_spec.cr#L110-L111